nix/status-go/default.nix

```diff
@@ -1,34 +1,50 @@
-{ target-os, stdenv, callPackage,
+{ target-os, config, stdenv, callPackage,
   buildGoPackage, go, fetchFromGitHub, openjdk,
   androidPkgs, xcodeWrapper }:
 
 let
   inherit (stdenv.lib)
     catAttrs concatStrings fileContents importJSON makeBinPath
-    optional optionalString removePrefix strings attrValues mapAttrs;
- 
+    optional optionalString removePrefix strings attrValues mapAttrs attrByPath
+    traceValFn;
+
   platform = callPackage ../platform.nix { inherit target-os; };
   utils = callPackage ../utils.nix { inherit xcodeWrapper; };
   gomobile = callPackage ./gomobile { inherit (androidPkgs) platform-tools; inherit target-os xcodeWrapper utils buildGoPackage; };
   buildStatusGoDesktopLib = callPackage ./build-desktop-status-go.nix { inherit buildGoPackage go xcodeWrapper utils; };
   buildStatusGoMobileLib = callPackage ./build-mobile-status-go.nix { inherit buildGoPackage go gomobile xcodeWrapper utils; };
-  versionJSON = importJSON ../../status-go-version.json; # TODO: Simplify this path search with lib.locateDominatingFile
-  versionRegex = "^v?[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+(-[[:alnum:].]+)$";
-  owner = versionJSON.owner;
-  repo = versionJSON.repo;
-  version = versionJSON.version;
-  sha256 = versionJSON.src-sha256;
-  rev = versionJSON.commit-sha1;
-  shortRev = strings.substring 0 7 rev;
-  goPackagePath = "github.com/${owner}/${repo}";
-  src = fetchFromGitHub { inherit rev owner repo sha256; name = "${repo}-${shortRev}-source"; };
-  # Replace src value with the path to a local status-go repository if you want to perform a build against it, e.g.
-  #src = /home/<user>/go/src/github.com/status-im/status-go;
+  srcData =
+    # If config.status_go.src_override is defined, instruct Nix to use that path to build status-go
+    if (attrByPath ["status_go" "src_override"] "" config) != "" then rec {
+        owner = "status-im";
+        repo = "status-go";
+        version = "local";
+        goPackagePath = "github.com/${owner}/${repo}";
+        rev = "unknown";
+        shortRev = "unknown";
+        versionName = "local";
```

Comment:
Why use `unknown` or `local`? We are already using `develop` for development versions, use that.

Reply:
What if the user is on a different branch? It felt weird using `develop` if my branch is named otherwise, that's why I chose a completely different hardcoded name. WDYT?

Reply:
Well, that just means in metrics we'll have two versions: `vunknown` and `vdevelop`.

In both cases the exact commit or branch is unknown, but they appear separately, is that a good thing?

Reply:
why would the `v` prefix be there? I think it might actually be a good thing to know that local test builds are not taken into account in `develop`. Or not?

Reply:
Because Geth code adds the `v` because it's retarded.

But `develop` also includes local builds, just ones that were built using commit hash instead of a custom `status-go` path, both are local, both are development builds. What's the difference?

Reply:
> Because Geth code adds the v because it's retarded.

Isn't that what you're removing with `then removePrefix "v" version # Geth forces a 'v' prefix`? So it looks like whatever we assign to `versionName` will be used as-is.

In any case, I don't care much either way, so I'll just change `versionName` to `develop`.

Reply:
No, that part removes `v` in case version contains `v`, so we don't get `vvSomething`.